README.Rmd

```diff
@@ -52,16 +52,26 @@ ellmer supports a wide variety of model providers:
 * Snowflake Cortex: `chat_snowflake()` and `chat_cortex_analyst()`.
 * VLLM: `chat_vllm()`.
 
-## Model choice
+### Provider/model choice
 
-If you're using ellmer inside your organisation, you'll be limited to what your IT department allows, which is likely to be one provided by a big cloud provider, e.g. `chat_azure()`, `chat_bedrock()`, `chat_databricks()`, or `chat_snowflake()`. If you're using ellmer for your own exploration, you'll have a lot more freedom, so we have a few recommendations to help you get started:
+If you're using ellmer inside an organisation, you may have internal policies that limit you to models from big cloud providers, e.g. `chat_azure()`, `chat_bedrock()`, `chat_databricks()`, or `chat_snowflake()`.
+
+If you're using ellmer for your own exploration, you'll have a lot more freedom, so we have a few recommendations to help you get started:
 
 - `chat_openai()` or `chat_claude()` are good places to start. `chat_openai()` defaults to **GPT-4o**, but you can use `model = "gpt-4o-mini"` for a cheaper, lower-quality model, or `model = "o1-mini"` for more complex reasoning. `chat_claude()` is also good; it defaults to **Claude 3.5 Sonnet**, which we have found to be particularly good at writing code.
 
 - `chat_gemini()` is great for large prompts because it has a much larger context window than other models. It allows up to 1 million tokens, compared to Claude 3.5 Sonnet's 200k and GPT-4o's 128k. It also comes with a generous free tier (with the downside that [your data is used](https://ai.google.dev/gemini-api/terms#unpaid-services) to improve the model).
 
 - `chat_ollama()`, which uses [Ollama](https://ollama.com), allows you to run models on your own computer. While the biggest models you can run locally aren't as good as the state of the art hosted models, they don't share your data and are effectively free.
 
+### Authentication
+
+Authentication works a little differently depending on the provider. A few popular ones (including OpenAI and Anthropic) require you to obtain an API key. We recommend you save it in an environment variable rather than using it directly in your code, and if you deploy an app or report that uses ellmer to another system, you'll need to ensure that this environment variable is available there, too.
+
+ellmer also automatically detects many of the OAuth or IAM-based credentials used by the big cloud providers (currently `chat_azure()`, `chat_bedrock()`, `chat_databricks()`, `chat_snowflake()`). That includes credentials for these platforms managed by Posit Workbench and Posit Connect.
```

Comment:
Overall, this seems like a big improvement, thanks! I think the bit that we're missing is some brief note about how to use viewer based auth on connect. Maybe something like:

> Posit connect also supports viewer-based auth, where the credential of the person using the app are used. To opt-in to this, you need to ...

Reply:
I guess for Connect, it's actually opt-out? i.e. if you don't want to use viewer-based auth, you need to set up a service account and use that? Do we have advice on that?

Reply:
Or do you need to do something on the server side too? so we should just link to https://docs.posit.co/connect/user/oauth-integrations/#adding-oauth-integrations-to-deployed-content

Reply:
For now, I'll just link. We can improve the linked docs if needed.